src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -109,20 +110,23 @@ public String displayMedia(String regexURL, String userInput) {
     Pattern pattern = Pattern.compile(regexURL);
     Matcher matcher = pattern.matcher(userInput);
 
+    List<String> imageExtensions = Arrays.asList("png ", "jpg ", "bmp ", "gif ", "svg ");
```

Comment:
hmm what if the url is the last part of the input, that way there will not be a space?

maybe remove the spaces here, but call mediaURL.trim() which removes any surrounding whitespaces (try it out in a console to see how it works)

Reply:
ohh yes, was thinking how to remove the spaces and trim() really helped! Now I have included support for newlines and endlines in the regex and also trimmed the input before searching the the extension ends. But now I noticed a bug(?) which is I am unable to put several same urls as the replace will replace all, replace first also limits to the first url so also doesn't work. Do you happen to have any suggestions for this?

Reply:
Hmmm not sure exactly what you mean, do you have an example and what it produces (wrongly)?

Reply:
yep! For example, if I were to put the link [](https://thumbs.gfycat.com/EqualAcidicFlies-max-1mb.gif) 3 times in a row, each time it will be replaced each time it is found to give `<img src=<img src=<img src=https://thumbs.gfycat.com/EqualAcidicFlies-max-1mb.gif /> /> />` and hence will not display correctly.